### PR TITLE
MTG-780 Optional collection auth map

### PR DESCRIPTION
**Pull request by n00m4d**
# What
This PR is one more synchronizer refactor work. It adds optional hashMap to get_asset_indexes method. That map contains collection key and authority value. It's necessary only for core assets.

# Why
It should decrease number of times we query DB during asset batch processing. For full synchronization process it can save us hours(potentially).

### CODE REVIEW

File: rocks-db/src/batch_client.rs [hunk 1]
```diff
@@ -142,13 +146,29 @@ impl AssetIndexReader for Storage {
             .iter()
             .flat_map(|c| c.as_ref().map(|c| c.collection.value))
             .collect::<Vec<_>>();
```
Comment (StanChe) [inline rocks-db/src/batch_client.rs:148]: You need to collect those pks only when you don't have the collection_authorities passed in.

File: rocks-db/src/batch_client.rs [hunk 2]
```diff
@@ -142,13 +146,29 @@ impl AssetIndexReader for Storage {
...
+                    .into_iter()
+                    .flatten()
+                    .filter_map(|asset| asset.authority.value.map(|v| (asset.pubkey, v)))
+                    .collect::<HashMap<_, _>>()
+            }
+        };
+
+        // mpl_core_map.is_empty() && collection_authorities.is_some() check covers case when DB is empty
+        let mpl_core_collections = if mpl_core_map.is_empty() && collection_authorities.is_some() {
+            collection_authorities.unwrap()
```
Comment (StanChe) [inline rocks-db/src/batch_client.rs:168]: why not return it above? like 
```
let mpl_core_collections = collection_authorities.unwrap_or_else(||self.asset_collection_data....)
```

File: rocks-db/src/dump_client.rs [hunk 3]
```diff
@@ -199,6 +200,25 @@ impl Storage {
             synchronizer_metrics.clone(),
         ));
 
+        let mut collections = HashMap::new();
```
Comment (StanChe) [inline rocks-db/src/dump_client.rs:203]: we need only the core collections, this way you may save 99.99% of the memory for this map

Comment (StanChe) [reply rocks-db/src/dump_client.rs:203]: so the suggested approach is 
```
let pubkeys = self.asset_static_data.iter_start().filter_map(|a|a.specification_asset_class == MplCoreCollection? Some(a.pubkey):none).collect;
let collections = self.asset_collection_data
            .batch_get(pubkeys)
```

**Comment by RequescoS** [comment]
Could we also add this HashMap to our API? I think it can increase the speed of requests, especially for methods that return many assets

**Review by StanChe — APPROVED** [review]

**Review by RequescoS — APPROVED** [review]

**Pull request merged**